webapp/src/test/java/com/box/l10n/mojito/monitoring/QuartzPendingJobsReportingTaskTest.java

```diff
@@ -0,0 +1,196 @@
+            builder = TriggerBuilder.newTrigger()
+                                    .forJob(job)
+                                    .startAt(Date.from(ZonedDateTime.now().plusHours(i+1).toInstant()));
+            scheduler.scheduleJob(job, builder.build());
+        }
+
+        task.reportPendingJobs();
+
+        RequiredSearch test1Gauges = meterRegistry.get("quartz.pending.jobs")
+                                                  .tag("jobClass", "QuartzPendingJobsReportingTaskTest$Test1Job")
+                                                  .tag("jobGroup", DYNAMIC_GROUP_NAME);
+
+        RequiredSearch test2Gauges = meterRegistry.get("quartz.pending.jobs")
+                                                  .tag("jobClass", "EmptyTestJob")
+                                                  .tag("jobGroup", DYNAMIC_GROUP_NAME);
+
+        await.untilAsserted(() -> assertThat(test1Gauges.gauges()).isNotEmpty());
+        await.untilAsserted(() -> assertThat(test2Gauges.gauges()).isNotEmpty());
+        await.untilAsserted(() -> assertThat(test1Gauges.gauge().value()).isEqualTo(test1Count));
+        await.untilAsserted(() -> assertThat(test2Gauges.gauge().value()).isEqualTo(emptyCount));
+    }
+
+    @Test
+    public void testFetchResults() throws Exception {
+
+        JobDetail job;
+        TriggerBuilder<Trigger> builder;
+        int test1Count = random.nextInt(5) + 2;
+        int emptyCount = random.nextInt(5) + 2;
+
+        for (int i = 0; i < test1Count; i++) {
+            job = newJob(Test1Job.class).withIdentity("Test1Job_" + i, DYNAMIC_GROUP_NAME).build();
+            builder = TriggerBuilder.newTrigger()
+                                    .forJob(job)
+                                    .startAt(Date.from(ZonedDateTime.now().plusHours(i+1).toInstant()));
+            scheduler.scheduleJob(job, builder.build());
+        }
+
+        for (int i = 0; i < emptyCount; i++) {
+            job = newJob(EmptyTestJob.class).withIdentity("EmptyJob_" + i, DYNAMIC_GROUP_NAME).build();
+            builder = TriggerBuilder.newTrigger()
+                                    .forJob(job)
+                                    .startAt(Date.from(ZonedDateTime.now().plusHours(i+1).toInstant()));
+            scheduler.scheduleJob(job, builder.build());
+        }
+
+        List<QuartzPendingJobsReportingTask.PendingJob> pendingJobs = task.fetchResults();
+
+        assertThat(pendingJobs).hasSize(2);
+        assertThat(pendingJobs).extracting("jobClass")
+                               .containsExactlyInAnyOrder("QuartzPendingJobsReportingTaskTest$Test1Job", "EmptyTestJob");
+        assertThat(pendingJobs).extracting("jobGroup")
+                               .contains(DYNAMIC_GROUP_NAME);
+        assertThat(pendingJobs).extracting("count")
+                               .containsExactlyInAnyOrder((long) test1Count, (long) emptyCount);
+    }
+
+    @Test
+    public void testExtractClassName(){
```

Comment:
indentation is weird again here.

`task.extractClassName("."))` may fail? i think that why I first asked for a test but then removed my comment because I supposed it is a edge case